commons/src/store/mailbox.ts

```diff
@@ -228,6 +228,25 @@ function initializeThreads() {
             }
             return { ...threads };
           });
+        } else {
+          // This happens when someone sends a message using composer and we want
+          // to update the existing thread with the sent message
+          update((threads) => {
+            if (incomingMessage.thread_id) {
+              let threadToUpdate = threads[queryKey][currentPage].threads.find(
+                (thread) => thread.id === foundThread.id,
+              );
+
+              if (threadToUpdate) {
+                const messages = foundThread.messages;
+                messages.push(incomingMessage);
+                foundThread.messages = messages;
+                foundThread.snippet = incomingMessage.snippet;
+                threadToUpdate = JSON.parse(JSON.stringify(foundThread));
+              }
+            }
+            return { ...threads };
```

Comment:
I think `threadsMap` might need to be updated in this block as well.

Reply:
I think it gets updated by reference in `#210`